backend/msgpack/queue.go

```diff
@@ -155,8 +223,10 @@ func (q *queue) reportQueueSize(reportInterval time.Duration) {
 
 type queueMetrics struct {
 	queueLen          tally.Histogram
+	queueSuccesses    tally.Counter
```

Comment:
nit: might be more clear to do
```
enqueueSuccesses tally.Counter
enqueueDropOldest tally.Counter
enqueueDropCurrent tally.Counter
```
?
You can then do `enqueueScope := scope.Tagged(map[string]string{"action":"enqueue"})` and reuse it for all three counters.